docs/extension.md

```diff
@@ -3,13 +3,289 @@
+            [new Variable('myVar', new StringType())]
+        );
+    }
+}
+```
+
+### Class template resolvers
+
+You can also create class template resolvers are used to resolve templates in context of class with custom logic different from method/standalone reolvers.
+
+You can create your own class template resolver by extending [`AbstractClassTemplateResolver`](../src/LatteTemplateResolver/AbstractTemplateResolver.php).
+
+```php
+use Efabrica\PHPStanLatte\LatteContext\LatteContext;
+use PHPStan\BetterReflection\Reflection\ReflectionClass;
+use PHPStan\Rules\RuleErrorBuilder;
+
+abstract class MyTemplateResolver extends AbstractClassTemplateResolver
+{
+    protected function getClassResult(ReflectionClass $reflectionClass, LatteContext $latteContext): LatteTemplateResolverResult
+    {
+        // here write your own logic
+    }
+}
+```
+
+### Node template resolvers
+
+If you need to resolve templates of different PHP parser node than class you can create your own node template resolver by implementing [`NodeLatteTemplateResolverInterface`](../src/LatteTemplateResolver/NodeLatteTemplateResolverInterface.php).
+
+```php
+use Efabrica\PHPStanLatte\Collector\CollectedData\CollectedResolvedNode;
+use Efabrica\PHPStanLatte\LatteContext\LatteContext;
+use Efabrica\PHPStanLatte\LatteContext\Resolver\NodeLatteContextResolverInterface;
+use Efabrica\PHPStanLatte\Template\TemplateContext;
+use Efabrica\PHPStanLatte\Template\Variable;
+use PhpParser\Node;
+use PHPStan\Analyser\Scope;
+use PHPStan\Node\InClassNode;
+use PHPStan\Type\StringType;
+
+abstract class AbstractClassTemplateResolver implements NodeLatteTemplateResolverInterface
+{
+    private const PARAM_CLASS_NAME = 'className';
+
+    // this method runs in initial phase of resolving templates for every parsed node and collects resolved nodes with some parameters
+    public function collect(Node $node, Scope $scope): array
+    {
+        if(!$node instanceof Node\Stmt\ClassMethod) {
+            return [];
+        }
+        return [
+            new CollectedResolvedNode(
+                static::class, // always pass current class here
+                $scope->getFile(), // declare file where you want to run latte context collectors
+                [self::PARAM_CLASS_NAME => $className] // pass any primitive type parameters to resolve method
+            )
+        ];
+    }
+
+    // this method runs after all nodes and latte context colllectors finishes
+    public function resolve(CollectedResolvedNode $resolvedNode, LatteContext $latteContext): LatteTemplateResolverResult
+    {
+        $result = new LatteTemplateResolverResult();
+        // add template with given context to set of analysed templates
+        $result->addTemplate(new Template(
+            'template/path.latte', // path to analysed template
+            Control::class, // actual class passed to node visitors
+            'resolved', // name of action (will show in error message)
```

Comment:
is this somehow used when path is collected with custom resolver? I think I had some problem with it - I've created my resolver but extension still shows error that it can't resolve template for Control::resolved method

Reply:
I do not understand what you mean. Can you show me code of your resolver and what is expected behaviour?

Reply:
I have to check it first. I’ll write you on Slack